WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController.swift

```diff
@@ -862,12 +868,34 @@ private extension NotificationsViewController {
         NotificationSyncMediator()?.markAsRead(note)
     }
 
+    /// Marks all messages as read under the selected filter.
+    ///
+    @objc func markAllAsRead() {
+        guard let notes = tableViewHandler.resultsController.fetchedObjects as? [Notification] else {
+            return
+        }
+
+        let unreadNotifications = notes.filter {
+            !$0.read
+        }
+
+        NotificationSyncMediator()?.markAsRead(unreadNotifications, completion: { error in
+            let notice = Notice(
+                title: error != nil ? Localization.markAllAsReadNoticeFailure : Localization.markAllAsReadNoticeSuccess,
+                message: nil
+            )
+            ActionDispatcherFacade().dispatch(NoticeAction.post(notice))
+        })
+        updateMarkAllAsReadButton()
```

Comment:
🤔 pretty sure `updateMarkAllAsReadButton` should go inside the `completion` block as well since the button would need updating after the notifications are updated.